libby_coy/week_04/calculator/calculator.rb

```diff
@@ -0,0 +1,132 @@
+# create methods for all the things you want the user to do
+
+def get_user_choice
+  # Show user the available operations
+  puts "(+) - Addition"
+  puts "(-) - Subtractionn"
+  puts "(*) - Multiplication"
+  puts "(/) - Division"
+  puts "(**) - Squaring"
+  puts "(<3) - BMI Calculator"
+  puts "(<>) - Trip Calculator"
+  puts "(q) - Quits"
+
+  print "Enter your selection: "
+  gets.downcase.chomp # Implicit return
+end
+
+def addition
+  print "Enter first number: "
+  first_num = gets.to_i
+
+  print "Enter second number: "
+  second_num = gets.to_i
+
+  print "The answer is #{first_num + second_num} \n"
+end
+
+def subtraction
+  print "Enter first number: "
+  first_num = gets.to_i
+
+  print "Enter second number: "
+  second_num = gets.to_i
+
+  print "The answer is #{first_num - second_num} \n"
+end
+
+def division
+  print "Enter first number: "
+  first_num = gets.to_i
+
+  print "Enter second number: "
+  second_num = gets.to_i
+
+  print "The answer is #{first_num / second_num} \n"
+end
+
+def multiplication
+  print "Enter first number: "
+  first_num = gets.to_i
+
+  print "Enter second number: "
+  second_num = gets.to_i
+
+  print "The answer is #{first_num * second_num} \n"
+end
+
+def squaring
+  print "Enter your number: "
+  first_num = gets.to_i
+
+  print "Enter the number to square it by: "
+  second_num = gets.to_i
+
+  print "The answer is #{first_num ** second_num} \n"
+end
+
+def body_mass_index
+  print "Enter your weight in kg: "
+  weight = gets.to_f
+
+  print "Enter your height in cm: "
+  height = (gets.to_f)/100
+
+  print "The answer is #{weight / (height * height)} \n"
+end
+
+def trip_calculator
+  print "Enter your total trip distance in miles: "
+  distance = gets.to_f
+
+  print "Enter your car's miles per gallon: "
+  mpg = gets.to_f
+
+  print "Enter the price per gallon in dollars: "
+  price = gets.to_f
+
+  print "Enter how many miles you will drive per hour: "
+  per_hour = gets.to_f
+
+  print "The trip will take you #{ distance / per_hour } hours\nand it will cost you $#{ distance / mpg * price }.\n "
+
```

Comment:
This is all fine, but I just wanted to point out that on line 91 and also 75, where you've done a calculation inside the string interpolation: once the calculations get more complex, it's probably better to move them out of the string and onto their own line of code:

```ruby
trip_cost = distance / mpg * price
trip_hours = distance / per_hour
print "The trip will take you #{ trip_hours } hours\nand it will cost you $#{ trip_cost }.\n"
```

It's just clearer, ultimately, and it will also be easier to add comments to your calculation lines to explain what you're doing with them.